app/src/main/java/com/khaithu/a3ctech_android_final_project/adapter/WatchlistRecyclerAdapter.kt

```diff
@@ -0,0 +1,73 @@
+package com.khaithu.a3ctech_android_final_project.adapter
+
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.khaithu.a3ctech_android_final_project.R
+import com.khaithu.a3ctech_android_final_project.helper.Constant
+import com.khaithu.a3ctech_android_final_project.helper.GlideHelper
+import com.khaithu.a3ctech_android_final_project.model.WatchlistItem
+import kotlinx.android.synthetic.main.item_movies.view.*
+import kotlinx.android.synthetic.main.item_movies.view.imageMovie
+import kotlinx.android.synthetic.main.item_movies.view.releaseDate
+import kotlinx.android.synthetic.main.item_movies.view.titleMovie
+import kotlinx.android.synthetic.main.item_watchlist.view.*
+
+class WatchlistRecyclerAdapter : RecyclerView.Adapter<WatchlistRecyclerAdapter.ViewHolder>() {
+
+    private lateinit var mIWatchListEvent: IWatchListEvent
+    private var lists: MutableList<WatchlistItem> = arrayListOf()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val inflater: LayoutInflater = LayoutInflater.from(parent.context)
+
+        val view: View = inflater.inflate(R.layout.item_watchlist, parent, false)
+
+        return ViewHolder(view, mIWatchListEvent)
+    }
+
+    override fun getItemCount(): Int {
+        return lists.size
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bind(lists[position])
+    }
+
+    fun updateData(lists: MutableList<WatchlistItem>) {
+        this.lists = lists
+        notifyDataSetChanged()
+    }
+
+    fun initEvent(handlerEvent : IWatchListEvent) {
+        this.mIWatchListEvent = handlerEvent
+    }
+
+    class ViewHolder(itemView: View,var mIWatchListEvent: IWatchListEvent) : RecyclerView.ViewHolder(itemView) {
+        fun bind(movie: WatchlistItem) {
+            itemView.titleMovieWatchlist.text = movie.title
+            itemView.releaseDateWatchlist.text = movie.releaseDate
+
+            GlideHelper.loadImage(
+                itemView,
+                Constant.posterBaseUrl + movie.posterImage,
+                itemView.imageMovieWatchlist
+            )
+
+            itemView.imageMovieWatchlist.setOnClickListener(View.OnClickListener {
+               mIWatchListEvent.handleEvent(movie.id)
+            })
+
+            itemView.textRemove.setOnClickListener(View.OnClickListener {
+                mIWatchListEvent.handleEventRemove(movie.id)
+            })
+        }
+    }
+}
+interface IWatchListEvent {
+    fun handleEvent(id :Int?)
+
+    fun handleEventRemove(id : Int?)
+}
```

Comment:
You should define clearly name of method. Easy to maintain later
My suggestion: 
- Name of callback class should be ended with Listener, example `MovieWatchlistListener`
- Name of callback method should be started with prefix `on`
`fun onMovieWatchlistClicked(id: Int?)`
`fun onMovieWatchlistRemoved(id: Int?)`
*pls format code this file*